### DISCUSSION TITLE
Using dictionary to assign colors to road types?

### DISCUSSION THREAD

**Discussion by peterkobylarz** [earthlab-education/Earth-Analytics-2023-01-Intro#503] [Q&A06 Vector Data] [answered]
I'm having an issue trying to symbolize the different road types in the site map notebook. I've created a dictionary that specifies a different color for each road type. However, when I run my code, all of the road types end up symbolizing the same color. Here is my code:

```
madera_plot = sjer_plot_gdf.hvplot()

roads_symb = {'Unknown': 'grey', 'M': 'blue', 'S': 'orange'}

for rttyp, gdf in sjer_road_clip_gdf.groupby('RTTYP'):
        print(rttyp)
        print(roads_symb[rttyp])
        color = roads_symb[rttyp]
        madera_plot *= sjer_road_clip_gdf.hvplot(line_color=color, label=rttyp)

madera_plot
```
Here's the output plot:
![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143359014/15bc7faa-fafb-404d-b814-66692e6b0d4b)

All roads are being symbolized as grey which is the final color/type in the for loop.

Any help would be appreciated!

**Comment by peterkobylarz** [comment]
I was able to come up with a work around that worked. In the for loop, I created a new geodataframe by querying the roads dataframe by type.

```for rttyp, gdf in sjer_road_clip_gdf.groupby('RTTYP'):
        print(rttyp)
        print(roads_symb[rttyp])
        road_gdf = sjer_road_clip_gdf[sjer_road_clip_gdf.RTTYP == rttyp]
        color = roads_symb[rttyp]
        madera_plot *= road_gdf.hvplot(line_color=color, label=rttyp)

madera_plot
```

I don't know if this is the most elegant solution but it allowed me to get the plot to symbolize correctly.

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143359014/a89744d7-632d-489e-9462-f8dc456ebdd8)

**Reply by peterkobylarz** [reply, marked as answer]
After thinking about this a little more, I think I found the correct solution. Rather than creating new GDFs for each subset of roads, I can just query the existing GDF for each iteration and plot using hvplot:

```
madera_plot *= (
        sjer_road_clip_gdf[sjer_road_clip_gdf.RTTYP == rttyp]
        .hvplot(line_color=color, label=rttyp)
)
```

**Comment by alexiemillikin** [comment]
Hi Peter,

I think the issue is that you call the specific "sjer_road_clip_gdf" within your for loop (bolded below, but in case that's hard to see it's in the second to last line), but you should just use "gdf", as that is what the loop is iterating over. I ran into a similar issue and this worked for me!

madera_plot = sjer_plot_gdf.hvplot()

roads_symb = {'Unknown': 'grey', 'M': 'blue', 'S': 'orange'}

for rttyp, gdf in sjer_road_clip_gdf.groupby('RTTYP'):
        print(rttyp)
        print(roads_symb[rttyp])
        color = roads_symb[rttyp]
        madera_plot *= **sjer_road_clip_gdf**.hvplot(line_color=color, label=rttyp)

madera_plot
